### PR TITLE
[WIP] MA-493 Return interfaces from API handling methods

### PR DESCRIPTION
**Pull request by njfaries**
# Description
WIP Commit. This is the start of work on changing a few API handlers to return interfaces instead of sending responses themselves. In the V2 case these still get formatted into a response using JSON serialization as the V2 requires that the response data be wrapped, which as we have seen does not work well with Magento's REST handlers.

This PR does not at present switch the V1 to also return an interface, as I wanted to get clarification on why it is currently structured the way that it is. See slack thread [here](https://boltpay.slack.com/archives/CQ864R9JQ/p1614726119112800)

~~Additionally, the tests are broken and I'm not quite sure how to resolve the issue of it complaining that I'm giving it a mocked instance of the `CreatedOrderInterfaceFactory`. See slack thread for this [here](https://boltpay.slack.com/archives/CQ864R9JQ/p1614729111115600)~~

Fixes: https://boltpay.atlassian.net/browse/MA-493

#changelog [WIP] MA-493 Return interfaces from API handling methods
Return interfaces from API handling methods

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Comment by danac-gs** [comment]
LGTM

File: Model/Api/UniversalApi.php [hunk 1]
```diff
@@ -125,10 +125,12 @@ public function execute(
             switch($event){
                 case "order.create":
                     //currently sends its own response updated for v2, no return
-                    $this->createOrder->execute(
-                        $event,
-                        isset($data['order']) ? $data['order'] : null,
-                        isset($data['currency']) ? $data['currency'] : null
+                    // $this->result->setData(
```
Comment (danac-gs) [inline Model/Api/UniversalApi.php:128]: remove redundant commented out code

**Review by danac-gs — APPROVED** [review]
LGTM

File: Model/Api/UniversalApi.php (hunk 1 above)
Comment (njfaries) [reply Model/Api/UniversalApi.php:128]: Oops, this got left commented out when I was experimenting with things. These lines should not be commented as `createOrder` now returns a value. Will fix.

**Comment by adentong** [comment]
this hasn't been touched in quite a while, so i'm closing it (somehow stale bot didn't close this)

File: Model/Api/CreateOrder.php [hunk 2]
```diff
@@ -296,7 +303,10 @@ public function execute(
                 ]]
             ]);
         } finally {
-            $this->response->sendResponse();
+            // we do not want to send a non-error response in the universal case. 
+            if (!$isUniversal) {
+                $this->response->sendResponse();
+            }
```
Comment (danac-gs) [inline Model/Api/CreateOrder.php:309]: Perhaps each of the exception catch above should be handled differently if $isUniversal == true

Comment (njfaries) [reply Model/Api/CreateOrder.php:309]: Potentially yes, that probably makes sense. Going to leave this closed for now though, the refactoring effort was more than anticipated and there were other things with priority. Will come back to this in future.

**Pull request closed without merging**